azurelinuxagent/daemon/main.py

```diff
@@ -69,6 +69,7 @@ def run(self, child_args=None):
         self.initialize_environment()
 
         CGroupConfigurator.get_instance().create_agent_cgroups(track_cgroups=False)
+        CGroupConfigurator.get_instance().cleanup_old_cgroups()
```

Comment:
since the cleanup is just temporary we should move the logic to create_agent_cgroups instead of calling a separate function

Reply:
I'm in favour of keeping it separate to maintain one responsibility per function and to make the testing easier. The removal logic stays the same wherever the call happens.